### PR TITLE
[Feature][core] support read arrow data

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
support read arrow data,  convert arrow data to SeatunnelRow. At present,  `doris` connector and `starrocks` connector use arrow mode to read data and extract this part of logic.

### Does this PR introduce _any_ user-facing change?
doris connector source ,  starrocks connector source

### How was this patch tested?
UT： Arrow2SeatunnelRowReaderTest#testSeatunnelRow
         Arrow2SeatunnelRowReaderTest#testConvertArrowSpeed

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/arrow/converter/BigIntConverter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.api.arrow.converter;
+
+import org.apache.seatunnel.shade.org.apache.arrow.vector.BigIntVector;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.types.Types;
+
+public class BigIntConverter implements Converter<BigIntVector> {
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/arrow/converter/BigIntConverter.java]: I think move to `connector-common` module for now would be better.

Comment (hawk9821) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/arrow/converter/BigIntConverter.java]: get

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/UInt4Converter.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.apache.seatunnel.connectors.seatunnel.common.source.arrow.converter;
+
+import org.apache.seatunnel.shade.org.apache.arrow.vector.UInt4Vector;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.types.Types;
+
+public class UInt4Converter implements Converter<UInt4Vector> {
+    @Override
+    public Object convert(int rowIndex, UInt4Vector fieldVector) {
+        return fieldVector.isNull(rowIndex) ? null : fieldVector.getObject(rowIndex);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/UInt4Converter.java]: Can we use an NullableConverter to wrap all converter to precheck the value is not null? I find all converter has null check.

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.function.Function;
+
+@Slf4j
+public class Arrow2SeatunnelRowReader implements AutoCloseable {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java]:
```suggestion
public class ArrowToSeatunnelRowReader implements AutoCloseable {
```

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.apache.seatunnel.connectors.seatunnel.common.source.arrow.converter;
+
+import org.apache.seatunnel.shade.org.apache.arrow.vector.complex.StructVector;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.types.Types;
+
+public class StructConverter implements Converter<StructVector> {
+    @Override
+    public Object convert(int rowIndex, StructVector fieldVector) {
+        return fieldVector.isNull(rowIndex) ? null : fieldVector.getObject(rowIndex);
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java:34]: I find most of convert method invoke `StructVector::getObject` to get value. So why we need create so many converter class?

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+    public Arrow2SeatunnelRowReader(byte[] byteArray, SeaTunnelRowType seaTunnelRowType) {
+        this.seaTunnelDataTypes = seaTunnelRowType.getFieldTypes();
+        this.rootAllocator = new RootAllocator(Integer.MAX_VALUE);
+        this.arrowStreamReader =
+                new ArrowStreamReader(new ByteArrayInputStream(byteArray), rootAllocator);
+        for (int i = 0; i < seaTunnelRowType.getFieldNames().length; i++) {
+            fieldIndexMap.put(seaTunnelRowType.getFieldNames()[i], i);
+        }
+        ServiceLoader<Converter> load = ServiceLoader.load(Converter.class);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java]: why not use static load?

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/UInt4Converter.java (hunk 2 above)
Comment (hawk9821) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/UInt4Converter.java]: use design patterns to optimize if else logic and improve code readability.   each implementation class of the interface  `org.apache.arrow.vector.FieldVector` corresponds to one converter.

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java (hunk 5 above)
Comment (hawk9821) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java]: image
arrow currently has a total of 45 types, each type corresponds to a Vector, each Vector corresponds to a converter,  The SPI class loading method is a little more brief

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java (hunk 4 above)
Comment (hawk9821) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java:34]: each converter corresponds to an arrow type，it's a one-to-one correspondence.  The mapping between the source database type and arrow type is determined by the source database, so each arrow type needs to be processed to ensure that the source data type is correctly converted to the Seatunnel data type.

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java (hunk 5 above)
Comment (hawk9821) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java]: fixed

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java (hunk 4 above)
Comment (hawk9821) [reply seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/converter/StructConverter.java:34]: Provide a default converter

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**